backend/lucene/src/main/java/org/hibernate/search/backend/lucene/index/impl/LuceneIndexSearchTargetBuilder.java

```diff
@@ -31,14 +31,14 @@
 	private final SearchBackendContext searchBackendContext;
 
 	// Use LinkedHashSet to ensure stable order when generating requests
-	private final Set<LuceneIndexManager> indexManagers = new LinkedHashSet<>();
+	private final Set<LuceneIndexManagerImplementor> indexManagers = new LinkedHashSet<>();
```

Comment:
We are not consistent between Es and Lucene. See here: https://github.com/hibernate/hibernate-search/pull/1727/commits/deaa1adba34aea50211e69c5ea80c9df22243c7c#diff-e34a887279fa744ebd6fbd73ab2e553aR33 .

Is it something expected e.g. you use the `Impl` in Es and an `Implementor` interface in Lucene?

Reply:
This inconsistency is pre-existing: there's an (internal) interface for index managers on the Lucene side, but not on the Elasticsearch side.

On the Elasticsearch side the interface is clearly useless, because there's no reason to have multiple index manager implementations; worst case some *internal* components could have multiple implementations.

On the Lucene side, we talked about it, and I wanted to just remove the interface, but you were worried that we might need multiple index manager types in the future and thought it was better this way. Maybe there was more to it, but then I don't remember.

Currently the interface is only used in a single package, which is the package where its only implementation is defined, so the interface is not really useful.

Maybe I should just replace references to the interface with references to the implementation class, and we can reintroduce the interface later if necessary, once we know what exactly is generic to every implementation and what is specific to the "directory-based" index managers?

Note that in any case, since the "index manager" already implements SPIs defined in the engine, in my mind it already has one responsibility and should not assume another one. If we ever need index-scoped, backend-specific, internal operations to be exposed to other packages, I'd rather create separate objects, much like what we did with `org.hibernate.search.backend.lucene.index.impl.IndexingBackendContext` and `org.hibernate.search.backend.lucene.search.query.impl.SearchBackendContext`. It helps keep the inter-package dependencies straightforward and manageable.

Reply:
Ok, I added two commits just for that.

Reply:
Looks good.